File: server/php/src/Application/DTOs/Common/AddressDto.php
<?php

namespace OnlinePayments\ExampleApp\Application\DTOs\Common;

class AddressDto
{
    public ?string $firstName;
    public ?string $lastName;
    public ?string $country;
    public ?string $zip;
    public ?string $city;
    public ?string $street;

    public function __construct(
        ?string $firstName,
        ?string $lastName,
        ?string $country,
        ?string $zip,
        ?string $city,
        ?string $street
    ) {
        $this->firstName = $firstName;
        $this->lastName = $lastName;
        $this->country = $country;
        $this->zip = $zip;
        $this->city = $city;
        $this->street = $street;
    }

    public static function fromArray(array $data): self
    {
        return new self(
            $data['firstName'] ?? null,
            $data['lastName'] ?? null,
            $data['country'] ?? null,
            $data['zip'] ?? null,
            $data['city'] ?? null,
            $data['street'] ?? null
        );
    }

    public function toArray(): array
    {
        return [
            'firstName' => $this->firstName,
            'lastName' => $this->lastName,
            'country' => $this->country,
            'zip' => $this->zip,
            'city' => $this->city,
            'street' => $this->street,
        ];
    }
}

File: server/python/app/application/dtos/common/address_dto.py
from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel
from typing import Optional

class AddressDto(BaseModel):
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    first_name: Optional[str] = None
    last_name: Optional[str] = None
    city: Optional[str] = None
    country: Optional[str] = None
    street: Optional[str] = None
    zip: Optional[str] = None